src/utils/patchUrlMappings.ts

```diff
@@ -128,7 +130,28 @@ function attemptSetNodeSrc(node: Node, mappings: Mapping[]) {
   if (node instanceof HTMLElement && node.hasAttribute('src')) {
     const url = absoluteURL(node.getAttribute('src') ?? '');
     if (url.host === window.location.host) return;
-    node.setAttribute('src', attemptRemap({url, mappings}).toString());
+
+    if (node.tagName.toLowerCase() === 'script') {
+      // Scripts are a special case, and need to be wholly recreated since
+      // modifying a script tag doesn't refetch.
+      attemptRecreateScriptNode(node, {url, mappings});
+    } else {
+      node.setAttribute('src', attemptRemap({url, mappings}).toString());
+    }
+  }
+}
+
+function attemptRecreateScriptNode(node: HTMLElement, {url, mappings}: RemapInput) {
+  const newUrl = attemptRemap({url, mappings});
+  if (url.toString() !== newUrl.toString()) {
+    const newNode = document.createElement(node.tagName);
```

Comment:
I couldn't `node.clone()` here because script nodes have *internal state* on whether or not the node has started its fetch. See https://html.spec.whatwg.org/multipage/scripting.html and search the page for "already started", or [this StackOverflow answer](https://stackoverflow.com/questions/28771542/why-dont-clonenode-script-tags-execute).

Reply:
Looks great. Thanks for calling out how we got here. It may be worth leaving a comment to explain the motivation

Well done 👏🏻